### DISCUSSION TITLE
Is it possible to add classes and props as object?

### DISCUSSION THREAD

**Discussion by DamianSz2000** [zauberzeug/nicegui#561] [Q&A] [answered]
It starts to get really messy if If i add a lot of classes and props, can i provide them as object somehow so i can indent them freely?
Cause smth like this doesnt work

```py
with ui.column()
          .classes("w-full h-full")
          .props("blah blah"):
```

**Comment by falkoschindler** [comment, marked as answer]
There are a couple of ways to do this:

1. Use Python's line continuation `\`:

    ```py
    with ui.column() \
            .classes("w-full h-full") \
            .props("blah blah"):
        ...
    ```

2. Break the builder-pattern into multiple lines:
    
    ```py
    with ui.column() as column:
        column.classes("w-full h-full")
        column.props("blah blah")
        ...
    ```

3. Keep the classes and props in strings:

    ```py
    classes = "w-full h-full"
    props = "blah blah"
    with ui.column().classes(classes).props(props):
        ...    
    ```

Apart from that we are already working on an improved Tailwind API (https://github.com/zauberzeug/nicegui/pull/362). This will allow you keep the Tailwind style definition in an object and apply it to different objects, like:

```py
style = Tailwind().width('full').height('full')
with ui.column() as column:
    ...
style.apply(column)
```
